neo/Ledger/MemoryPool.cs

```diff
@@ -16,32 +16,33 @@ public class MemoryPool : IReadOnlyCollection<Transaction>
     {
         private class PoolItem : IComparable<PoolItem>
         {
-            public readonly Transaction Transaction;
+            public readonly Transaction Tx;
             public readonly DateTime Timestamp;
             public DateTime LastBroadcastTimestamp;
 
             public PoolItem(Transaction tx)
             {
-                Transaction = tx;
+                Tx = tx;
                 Timestamp = DateTime.UtcNow;
                 LastBroadcastTimestamp = Timestamp;
             }
 
-            public int CompareTo(Transaction tx)
+            public int CompareTo(Transaction otherTx)
             {
-                if (tx == null) return 1;
-                int ret = Transaction.FeePerByte.CompareTo(tx.FeePerByte);
+                if (otherTx == null) return 1;
+                // Fees sorted ascending
+                int ret = Tx.FeePerByte.CompareTo(otherTx.FeePerByte);
                 if (ret != 0) return ret;
-                ret = Transaction.NetworkFee.CompareTo(tx.NetworkFee);
+                ret = Tx.NetworkFee.CompareTo(otherTx.NetworkFee);
                 if (ret != 0) return ret;
-
-                return Transaction.Hash.CompareTo(tx.Hash);
+                // Transaction hash sorted descending
+                return otherTx.Hash.CompareTo(Tx.Hash);
```

Comment:
If think this logic is a little strange here... It should be `Tx.Hash.CompareTo(otherTx.Hash)`.

Because x.CompareTo(y) implies that, if x ">" y, returns 1. Greater should be one, and I think this will become -1 here.
If you want to sort in opposite manner, perhaps it's better just to inform the pool that the sorting is descending or ascending.

Reply:
I think it was like you said for ascending, now it is descending.
I am not sure, the tests may help.

Reply:
the PoolItems are sorted ascending within the class. If we switch this to the other way arooud it sorts ascending and will be wrong when reversed.

Reply:
It cannot be Tx.Hash.CompareTo(otherTx.Hash) because then it would be how it was before (it would be treating higher hashes as higher priority). The pool items are sorted ascending in the sorted sets in MemoryPool with highest priority being at the end.

Reply:
As you want, with the tests verifying that no problem.
The Reverse is a little strange to me but the result if the important thing.
In terms of efficiency the reverse is not costly, right?